### PR TITLE
UCP/GTEST: Add ucp_mmap multithreaded test + remove lock in ucp_rkey_pack

### PR DESCRIPTION
**Pull request by petro-rudenko**
## What

- Added test_ucp_mmap_mt test
- Removed context lock in `ucp_rkey_pack`

## Why ?
image
After this PR:
image

TODO: remove lock from `ucp_mem_map`. Would be done in the next PR, wen tested

### CODE REVIEW

**Comment by petro-rudenko** [comment]
bot:pipe:retest

File: src/ucp/core/ucp_rkey.c [hunk 1]
```diff
@@ -117,9 +117,6 @@ ucs_status_t ucp_rkey_pack(ucp_context_h context, ucp_mem_h memh,
     UCP_CONTEXT_CHECK_FEATURE_FLAGS(context, UCP_FEATURE_RMA | UCP_FEATURE_AMO,
                                     return UCS_ERR_INVALID_PARAM);
 
-    /* always acquire context lock */
-    UCP_THREAD_CS_ENTER(&context->mt_lock);
```
Comment (yosefe) [inline src/ucp/core/ucp_rkey.c:121]: pls review uct_ib_mkey_pack, and other UCTs: they are not really thread safe.

Comment (petro-rudenko) [reply src/ucp/core/ucp_rkey.c:121]: They are really not thread safe, but they are per MD / MH. Adding corner test to do parallel rkey_pack for the same MH. The only place I found where it modifies MD is here: https://github.com/openucx/ucx/blob/master/src/uct/ib/mlx5/dv/ib_mlx5dv_md.c#L269

Comment (yosefe) [reply src/ucp/core/ucp_rkey.c:121]: but the mkey itself is modified during pack

**Pull request closed without merging**